src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,24 +1,39 @@
-# TO-DO: Complete the selection_sort() function below
+import time
+
+# * Complete the selection_sort() function below
+# ? - O(n²)
+
 def selection_sort(arr):
-    # loop through n-1 elements
+
+    # * loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc)
+        # * find next smallest element
+        # * (hint, can do in 3 loc)
         # Your code here
+        for j in range(cur_index + 1, len(arr)):
+            if arr[smallest_index] > arr[j]:
+                smallest_index = j
+
 
 
-        # TO-DO: swap
+        # * swap
         # Your code here
+        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]
 
     return arr
 
 
-# TO-DO:  implement the Bubble Sort function below
-def bubble_sort(arr):
-    # Your code here
+# *  implement the Bubble Sort function below
+# ? - O(n²)
 
+def bubble_sort(arr):
+    l = len(arr)
+    for i in range(l - 1):
+        for p in range(0, l - i - 1):
+            if arr[p] > arr[p + 1]:
+                arr[p], arr[p + 1] = arr[p + 1], arr[p]
 
     return arr
```

Comment:
This is absolutely a valid (and very pythonic, I might add) bubble sort implementation. 

Having said that, this implementation is `O(n^2)` in _every_ case, even if the list is already sorted when passed to the function. 

Can you think of a way to implement this that allows for the function to run in `O(n)`time in the best case scenario?

Reply:
oh yeah!! Just a sec, I remember this.